src/ui/QueryEditor.tsx

```diff
@@ -0,0 +1,152 @@
+import React from 'react';
+import { QueryEditorProps, SelectableValue } from '@grafana/data';
+import { DataSource } from '../DataSource';
+import { CmkQuery, DataSourceOptions, ResponseDataAutocomplete } from '../types';
+import { cloneDeep } from 'lodash';
+import { InlineFieldRow, VerticalGroup } from '@grafana/ui';
+import { FilterEditor, Select, StringSelect } from './components';
+import { createAutocompleteConfig, Presentation } from './autocomplete';
+import { defaultRequestSpec, RequestSpec } from '../RequestSpec';
+import { titleCase } from '../utils';
+
+type GraphKind = 'template' | 'metric';
+
+type Props = QueryEditorProps<DataSource, CmkQuery, DataSourceOptions>;
+
+export const QueryEditor = (props: Props): JSX.Element => {
+  const { onChange, onRunQuery, datasource, query } = props;
+  const requestSpec = query.requestSpec || defaultRequestSpec;
+
+  const editionMode = datasource.getEdition() ?? 'RAW';
+
+  function isMinimalRequest(spec: RequestSpec) {
+    return [spec.site, spec.host_name, spec.graph_type, spec.graph].map(Boolean).reduce((x, y) => x && y);
+  }
+
+  const autocomplete =
```

Comment:
should/could we move all autocompleter stuff into a separate file? the only dependency is autocompleterRequest, right?